oci/oci.go

```diff
@@ -14,53 +15,80 @@ import (
 	spec "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
-// Init initializes the img directory as an OCI image.
-// i.e. Creates oci-layout, index.json, and blobs.
-//
-// img directory must not exist before calling this function.
+// BlobWriter writes an OCI blob and returns a digest when closed.
+type BlobWriter interface {
```

Comment:
e.g. future `content` package may use non-filesystem blob store, or just use sharded directory (`blobs/sha256/de/deadbeef`)
Then it could not be deduplicated